### PR TITLE
Document types for persisting scan run result

### PR DESCRIPTION
**Pull request by lamaks**
#### Description of changes

Document types for persisting scan run result.

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: packages/storage-documents/src/page-scan-run-result-map.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+export declare type DocumentType = 'pageScanRunResult' | 'pageScanRunSarifReport';
+
+/**
+ * The mapping document between scan run ID and all scan run corresponding documents.
+ */
+export interface PageScanRunResultMap {
```
Comment (devaradhanm) [inline packages/storage-documents/src/page-scan-run-result-map.ts]: how is this different from PageScanRunResult? what adidtional info this has?

File: packages/storage-documents/src/page-scan-run-result-map.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    scanId: string;
+    url: string;
+    links: Link[];
```
Comment (haonliu) [inline packages/storage-documents/src/page-scan-run-result-map.ts]: 'link' sounds a bit too general?

File: packages/storage-documents/src/page-scan-run-result-map.ts (hunk 1 above)
Comment (lamaks) [reply packages/storage-documents/src/page-scan-run-result-map.ts]: This is mapping file located under lookup folder with links pointing to historical folder of an actual scan results

File: packages/storage-documents/src/page-scan-run-result-map.ts (hunk 2 above)
Comment (lamaks) [reply packages/storage-documents/src/page-scan-run-result-map.ts]: This is what it is - link to other files location

File: packages/storage-documents/src/page-scan-run-result.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { StorageDocument } from '.';
+
+export declare type ReportFormat = 'sarif';
+export declare type ScanState = 'unknown' | 'pass' | 'fail';
+export declare type RunState = 'unknown' | 'accepted' | 'queued' | 'running' | 'completed' | 'failed';
+
+/**
+ * The web page scan run result document.
+ */
+export interface PageScanRunResult extends StorageDocument {
```
Comment (haonliu) [inline packages/storage-documents/src/page-scan-run-result.ts]: we have same types defined in 2 packages?

Comment (lamaks) [reply packages/storage-documents/src/page-scan-run-result.ts]: The web-api interface is for client response that is decoupled from a storage version that is different and may change w/o affecting client REST API interface

File: packages/storage-documents/src/page-scan-run-result.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { StorageDocument } from '.';
+
+export declare type ReportFormat = 'sarif';
+export declare type ScanState = 'unknown' | 'pass' | 'fail';
+export declare type RunState = 'unknown' | 'accepted' | 'queued' | 'running' | 'completed' | 'failed';
+
+/**
+ * The web page scan run result document.
```
Comment (devaradhanm) [inline packages/storage-documents/src/page-scan-run-result.ts]: can you scope all the new interfaces under WebApiFlowDocuments or a new package to make it eaiser to understand where these are used?

**Pull request merged**